### PR TITLE
PEO-1179 | Added sentry

### PR DESCRIPTION
**Pull request by WoLkErSs**
[PEO-1179](https://garage.atlassian.net/browse/PEO-1179)

### Description

Added gem `sentry-raven` and default config to sending notifications.

### Before submitting the merge request make sure the following are checked

- [x] Followed the style guidelines of this project
- [x] Performed a self-review of own code
- [ ] Wrote the tests that prove that fix is effective/that feature works

### CODE REVIEW

File: app/workers/sentry_worker.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class SentryWorker
```
Comment (chubchenko) [inline app/workers/sentry_worker.rb]: Propsoe to create an `ApplicationWorker` and add `include Sidekiq::Worker` to it

File: app/workers/sentry_worker.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  include Sidekiq::Worker
+
+  sidekiq_options queue: :notifiers
```
Comment (chubchenko) [inline app/workers/sentry_worker.rb]: Propose to remove this configuration and use the default queue

File: spec/workers/sentry_worker_spec.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+RSpec.describe SentryWorker, type: :worker do
+  subject(:worker_subject) { described_class.new }
```
Comment (chubchenko) [inline spec/workers/sentry_worker_spec.rb]: Propose to rename this variable to `worker`

File: spec/workers/sentry_worker_spec.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  let(:event) { 'event' }
+
+  it 'adds job to a queue' do
+    described_class.perform_async(event)
+    expect(described_class).to have_enqueued_sidekiq_job(event)
+  end
+
+  it 'calls Raven.send_event with the correct arguments' do
+    allow(Raven).to receive(:send_event)
+    expect(Raven).to receive(:send_event).with(event)
```
Comment (chubchenko) [inline spec/workers/sentry_worker_spec.rb:15]: I suppose that this line should be after `#perform` call

**Review by chubchenko — CHANGES_REQUESTED** [review]

File: spec/workers/sentry_worker_spec.rb (hunk 4 above)
Comment (WoLkErSs) [reply spec/workers/sentry_worker_spec.rb:15]: This is logical, but the test fails if up

File: app/workers/sentry_worker.rb (hunk 1 above)
Comment (WoLkErSs) [reply app/workers/sentry_worker.rb]: Method `perform` unavailible when name is `ApplicationWorker` 
I have named `BaseWorker`

File: app/workers/base_worker.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class BaseWorker
```
Comment (chubchenko) [inline app/workers/base_worker.rb]: To follow RoR naming convention propose to rename it to `ApplicationWorker`

Also, this worker should not have any implementation, for sentry we need to have a separate worker which will be inherited from the `ApplicationWorker`

**Review by chubchenko — CHANGES_REQUESTED** [review]

**Comment by chubchenko** [comment]
@WoLkErSs could you please update PR description

File: app/workers/application_worker.rb [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
+
+class ApplicationWorker
```
Comment (chubchenko) [inline app/workers/application_worker.rb:4]: do we need to include Sidekiq::Worker?

File: spec/workers/sentry_worker_spec.rb (hunk 4 above)
Comment (chubchenko) [reply spec/workers/sentry_worker_spec.rb:15]: I mean next:
allow(Raven).to receive(:send_event)
worker.perform(event)
expect(Raven).to receive(:send_event).with(event)

**Review by chubchenko — CHANGES_REQUESTED** [review]

**Pull request merged**